File: frontend/src/microservices/operations/airline/objects/ManagerCabinPricingInput.ts
// 鏈枃浠跺畾涔?operations 妯″潡鐨?`ManagerCabinPricingInput`锛屼綔涓鸿緭鍏ユā鍨嬪苟鎻愪緵 JSON 缂栬В鐮併€?

export type ManagerCabinPricingInput = {
  seatCount: number
  originalPrice: string
  discounted: boolean
  discountRate: string
}

export const managerCabinPricingInputFromJson = (json: string): ManagerCabinPricingInput =>
  JSON.parse(json) as ManagerCabinPricingInput

export const managerCabinPricingInputToJson = (value: ManagerCabinPricingInput): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/airline/objects/ManagerCabinPricingInput.scala
// 这个文件定义 airline 航班创建时的舱位定价输入。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerCabinPricingInput(
    seatCount: Int,
    originalPrice: String,
    discounted: Boolean,
    discountRate: String
)
object ManagerCabinPricingInput:
  given sourceEncoder: Encoder[ManagerCabinPricingInput] = deriveEncoder
  given sourceDecoder: Decoder[ManagerCabinPricingInput] = deriveDecoder
